src/test/java/org/jenkinsci/plugins/badge/extensions/SpecialValueParameterResolverExtensionTest.java

```diff
@@ -73,4 +75,37 @@ void testMultipleParameters() {
                 extension.resolve(mockRun, "buildId buildNumber duration startTime displayName description");
         assertThat(actualParameter, is("1234 1234 23:35 23:35 display name run description"));
     }
+
+    @Test
+    void resolveEmptyStringParameter() {
+        String actualParameter = extension.resolve(mockRun, "");
+        assertThat(actualParameter, is(""));
+    }
+
+    @Test
+    void resolveWithNullParameter() {
+        String actualParameter = extension.resolve(mockRun, null);
+        assertThat(actualParameter, is((String) null));
+    }
+
+    @Test
+    void unknownParameterShouldNotBeResolved() {
+        String actualParameter = extension.resolve(mockRun, "unknown");
+        assertThat(actualParameter, is("unknown"));
+    }
+
+    @Test
+    void testResolveWithNoMatchingParameters() {
+        String actualParameter = extension.resolve(mockRun, "noMatchingParameters");
+        assertThat(actualParameter, is("noMatchingParameters"));
+    }
```

Comment:
This does not seem to test anything more than is already tested by `unknownParameterShouldNotBeResolved`.  I think it should be removed as a duplicate of `unknownParameterShouldNotBeResolved`
```suggestion
```